TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/HttpByteBufFormatter.java

```diff
@@ -172,6 +185,9 @@ public static HttpMessage parseHttpMessageFromBufs(HttpMessageType msgType, Stre
         });
 
         try {
+            if (throwable[0] != null) {
+                throw Lombok.sneakyThrow(throwable[0]);
+            }
             return channel.readInbound();
```

Comment:
I'm not convinced that the above is going to fix a leak.  Please connect with me off-line so that we can discuss.
I'm suspicious that the finally block for channel.finishAndReleaseAll() should be covering everything post the channel declaration.  I suspect that we have some stuff built within the channel, then throw before we go to check it out.